### ISSUE TITLE
`pl.coalesce()` gives wrong List dtype in lazy mode

### ISSUE THREAD

**Issue by ghuls** [pola-rs/polars#8415]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

`pl.coalesce()` gives wrong List dtype in lazy mode.

### Reproducible example

```python
In [33]: lf = pl.LazyFrame([pl.Series("a", [1, None, 3]), pl.Series("b", [None, 20, 30])])

In [34]: lf.schema
Out[34]: {'a': Int64, 'b': Int64}

In [35]: lf.with_columns(pl.coalesce(pl.col("a"), pl.col("b")).alias("c"))
Out[35]: <polars.LazyFrame object at 0x7FAB5D157E20>

In [36]: lf.with_columns(pl.coalesce(pl.col("a"), pl.col("b")).alias("c")).collect()
Out[36]: 
shape: (3, 3)
┌──────┬──────┬─────┐
│ a    ┆ b    ┆ c   │
│ ---  ┆ ---  ┆ --- │
│ i64  ┆ i64  ┆ i64 │
╞══════╪══════╪═════╡
│ 1    ┆ null ┆ 1   │
│ null ┆ 20   ┆ 20  │
│ 3    ┆ 30   ┆ 3   │
└──────┴──────┴─────┘

In [37]: lf.with_columns(pl.coalesce(pl.col("a"), pl.col("b")).alias("c")).collect().schema
Out[37]: {'a': Int64, 'b': Int64, 'c': Int64}

# Wrong dtype for column "c" in lazy with pl.coalesce
In [38]: lf.with_columns(pl.coalesce(pl.col("a"), pl.col("b")).alias("c")).schema
Out[38]: {'a': Int64, 'b': Int64, 'c': List(Int64)}
```

### Expected behavior

Eager and lazy give back the same schema when `pl.coalesce()` is used and pl.coalesce should not return a List dtype.

### Installed versions

<details>

```
---Version info---
Polars: 0.17.5
Index type: UInt32
Platform: Linux-5.15.0-69-generic-x86_64-with-glibc2.31
Python: 3.10.10 | packaged by conda-forge | (main, Mar 24 2023, 20:08:06) [GCC 11.3.0]
---Optional dependencies---
numpy: 1.24.2
pandas: 2.0.0
pyarrow: 11.0.0
connectorx: <not installed>
deltalake: <not installed>
fsspec: <not installed>
matplotlib: 3.7.1
xlsx2csv: <not installed>
xlsxwriter: <not installed>
```

</details>

**Comment by ritchie46** [comment]
I believe I just fixed this. Is it also the case on main?

**Comment by ghuls** [comment]
Yeah, it is fixed in main now.

Duplicate of #8324
Fixed by #8324
